operator/internal/manifests/internal/alerts/prometheus-alerts.yaml

```diff
@@ -82,7 +70,114 @@ groups:
+        sum(
+          irate(
+            loki_request_duration_seconds_bucket{route!~"(?i).*tail.*"}[1m]
+          )
+        ) by (le, job, namespace, route)
+      )
+      * 100
+      > 1
+    for: 15m
+    labels:
+      severity: critical
+  - alert: LokiTenantRateLimit
+    annotations:
+      message: |-
+        {{ $labels.job }} {{ $labels.route }} is experiencing 429 errors.
+      summary: "At least 10% of requests are responded with the rate limit error code."
+      runbook_url: "[[ .RunbookURL ]]#Loki-Tenant-Rate-Limit"
+    expr: |
+      sum(
+        code:loki_request_duration_seconds_count:irate1m{status_code="429"}
+      ) by (job, namespace, route)
+      /
+      sum(
+        code:loki_request_duration_seconds_count:irate1m
+      ) by (job, namespace, route)
+      * 100
+      > 10
+    for: 15m
+    labels:
+      severity: warning
+  - alert: LokiStorageSlowWrite
+    annotations:
+      message: |-
+        The storage path is experiencing slow write response rates.
+      summary: "The storage path is experiencing slow write response rates."
+      runbook_url: "[[ .RunbookURL ]]#Loki-Storage-Slow-Write"
+    expr: |
+      histogram_quantile(0.99,
+        sum(
+          operation:loki_boltdb_shipper_request_duration_seconds_bucket:rate5m{operation="WRITE"}
+        ) by (le, job, namespace)
+      )
+      * 100
+      > 1
+    for: 15m
+    labels:
+      severity: warning
+  - alert: LokiStorageSlowRead
+    annotations:
+      message: |-
+        The storage path is experiencing slow read response rates.
+      summary: "The storage path is experiencing slow read response rates."
+      runbook_url: "[[ .RunbookURL ]]#Loki-Storage-Slow-Read"
+    expr: |
+      histogram_quantile(0.99,
+        sum(
+          operation:loki_boltdb_shipper_request_duration_seconds_bucket:rate5m{operation="Shipper.Query"}
+        ) by (le, job, namespace)
+      )
+      * 100
+      > 5
+    for: 15m
+    labels:
+      severity: warning
+  - alert: LokiWritePathHighLoad
+    annotations:
+      message: |-
+        The write path is experiencing high load.
+      summary: "The write path is experiencing high load, causing backpressure storage flushing."
+      runbook_url: "[[ .RunbookURL ]]#Loki-Write-Path-High-Load"
+    expr: |
+      sum(
+        max_over_time(
+          loki_ingester_wal_replay_flushing[1m]
+        )
+      ) by (job, namespace)
+      > 0
+    for: 15m
+    labels:
+      severity: warning
+  - alert: LokiReadPathHighLoad
+    annotations:
+      message: |-
+        The read path is experiencing high load.
+      summary: "The read path has high volume of queries, causing longer response times."
+      runbook_url: "[[ .RunbookURL ]]#Loki-Read-Path-High-Load"
+    expr: |
+      histogram_quantile(0.99,
+        sum(
+          rate(
+            loki_logql_querystats_latency_seconds_bucket[5m]
+          )
+        ) by (le, job, namespace)
+      )
+      * 100
+      > 30
```

Comment:
Is this maybe a copy-paste error from the percent-based alerts? The unit of the `histogram_quantile` result should be "seconds".

Reply:
From the tests, the `LokiRequestLatency` alert, and looking at the values, I think the value does need to be multiplied by 100 to actually produce results.